const AppError = require("../utils/appError");
const catchAsync = require("../utils/catchAsync");
const axios = require("axios");

exports.getVulnerability = catchAsync(async (req, res, next) => {
    const url = "https://services.nvd.nist.gov/rest/json/cves/2.0?keywordSearch=PAX&keywordExactMatch&pubStartDate=2024-08-04T00:00:00.000&pubEndDate=2024-10-22T00:00:00.000";

    try {
        const response = await axios.get(url);

        if (response.status === 200) {
            res.status(200).json({
                status: 'success',
                data: response.data
            });
        } else {
            return next(new AppError(`Failed to fetch data: ${response.statusText}`, response.status));
        }
    } catch (error) {
        console.error("Error fetching vulnerability:", error);
        return next(new AppError('Error fetching vulnerability', 500));
    }
});